lib/Dancer2/Core/Error.pm

```diff
@@ -255,8 +254,13 @@ has content => (
                     }
                 );
             };
+
             # return rendered content unless there was an error.
-            defined $content && return $content;
+            if (defined $content) {
+                # Why is this necessary here and not with other templates e.g. like route templates?
+                utf8::encode($content) if utf8::is_utf8($content); # less confusing via String::UnicodeUTF8: $content = get_utf8($content) if is_unicode($content)
```

Comment:
You can configure an app to use an encoding other than UTF-8. The "Why is it necessary" comment and having to do that twice points to this being the wrong place to be dealing with encodings.

For the content returned from a route (including templates), that content is added to the response object then the response is asked to encode it via `$response->content($content); $response->encode_content;`.

`Dancer2::Core::Error->throw` populates the response. Do any encoding there.